plugins/out_stackdriver/stackdriver.c

```diff
@@ -2637,6 +2768,15 @@ static void cb_stackdriver_flush(struct flb_event_chunk *event_chunk,
         }
         else if (c->resp.status >= 400 && c->resp.status < 500) {
             ret_code = FLB_ERROR;
+            if (c->resp.status == 400){
+              ret = parse_partial_success_response(c, ctx, &partial_failure);
```

Comment:
Careful! You're overwriting `ret` from line 2713, which will change the behavior of line 2770.

Luckily, you'd have to rewrite that code anyway, because you have to call `update_http_metrics` multiple times, once with each of the error codes present in `"logEntryErrors"` (possibly translated to HTTP codes, though the fluentd plugin does not do that). See [this](https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/master/lib/fluent/plugin/out_google_cloud.rb#L1060-L1082) for reference.